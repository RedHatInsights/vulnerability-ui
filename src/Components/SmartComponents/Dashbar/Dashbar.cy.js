import React from 'react';
import { mount } from '@cypress/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import ReducerRegistry from '../../../Utilities/ReducerRegistry';
import dashbarResponse from '../../../../cypress/fixtures/dashbar.json';
import announcementResponse from '../../../../cypress/fixtures/announcement.json';
import { IntlProvider } from '@redhat-cloud-services/frontend-components-translations';
import messages from '../../../../locales/en.json';
import Dashbar from './Dashbar';
import { changeCveListParameters } from '../../../Store/Actions/Actions';

const store = ReducerRegistry.getStore();

beforeEach(() => {
    cy.intercept('GET', '**/api/vulnerability/v1/dashbar', dashbarResponse);
    cy.intercept('GET', '**/api/vulnerability/v1/announcement', announcementResponse);

    mount(
        <IntlProvider locale="en" messages={messages}>
            <Provider store={store}>
                <Router>
                    <Dashbar />
                </Router>
            </Provider>
        </IntlProvider>
    );
});

describe('Dashbar', () => {
    it('exists and matches screenshot', () => {
        store.dispatch(changeCveListParameters({}));

        cy.get('#dashbar').matchImage();
        cy.get('#announcement').matchImage();
    });
});
