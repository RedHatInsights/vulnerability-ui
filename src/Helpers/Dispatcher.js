import { addNotification } from '@red-hat-insights/insights-frontend-components/components/Notifications';
import ReducerRegistry from '../Utilities/ReducerRegistry';
export function dispatchAction(actionCreator) {
    const store = ReducerRegistry.getStore();
    return store.dispatch(actionCreator).catch(res => {
        if ('json' in res) {
            let parsedMessage = res.json();
            parsedMessage.then(arr => {
                for (let err of arr.errors) {
                    store.dispatch(
                        addNotification({
                            variant: 'danger',
                            title: 'Error ' + err.status,
                            description: err.detail
                        })
                    );
                }
            });
        } else {
            store.dispatch(
                addNotification({
                    variant: 'danger',
                    title: 'Error ' + res.status,
                    description: res.statusText
                })
            );
        }
    });
}
