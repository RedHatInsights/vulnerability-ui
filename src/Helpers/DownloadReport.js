import { downloadFile } from '@redhat-cloud-services/frontend-components-utilities/files/helpers';

class DownloadReport  {
    constructor() {
        this.defaultParams = {
            page: 1,
            page_size: Number.MAX_SAFE_INTEGER,
            date: new Date().toISOString().replace(/[T:]/g, '-').split('.')[0] + '-utc'
        };

        this.params = {
            cve: ''
        };

        this.pages = ['systems-exposed', 'system-cves', 'cves', 'system-list'];
        this.formats = ['csv', 'json'];
    }

    filename = (page) => {
        switch (page) {
            case 'systems-exposed':
                return `vulnerability_systems-exposed-${this.params.cve}--${this.defaultParams.date}`;
            case 'system-list':
                return `vulnerability_systems--${this.defaultParams.date}`;
            default:
                return `vulnerability_cves--${this.defaultParams.date}`;

        }
    }

    checkTypes = (fetchData, page, format) => {
        if (typeof fetchData !== 'function') {
            throw `exec expects the 1st param to be funtion. Received ${typeof fetchData}`;
        }

        if (!this.pages.includes(page)) {
            throw `Page ${page} does not exist`;
        }

        if (!this.formats.includes(format)) {
            throw `Format ${format} is not supported`;
        }
    }

    /**
     * Execute and download the report
     *
     * @param {Function} fetchData - Function to be called in order to fetch data.
     * @param {Object || Array} params - Object to be sent as request params. If you need to pass extra argument, pass an array.
     * @param {String} format - Supported formats to download are CSV and JSON.
     * @param {String} page - Name of the page based on which the filename will be determined.
     * @returns {void}
     */
    exec = async (fetchData, params, format, page) => {
        this.checkTypes(fetchData, page, format);
        this.params.cve = Array.isArray(params) ? params[0] : '';

        try {
            let { payload } = Array.isArray(params)
                ? await fetchData(this.params.cve, {
                    ...params[1],
                    ...this.defaultParams,
                    data_format: format
                }) :
                await fetchData({
                    ...params,
                    ...this.defaultParams,
                    data_format: format
                });

            let { data: response } = await payload;
            let data = format === 'json' ? JSON.stringify(response) : response;
            // downaloads the files
            downloadFile(data, this.filename(page), format);
        } catch (error) {
            throw `${error}`;
        }

    }

}

export default new DownloadReport();
