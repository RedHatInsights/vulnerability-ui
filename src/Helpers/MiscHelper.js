import { SecurityIcon, UnknownIcon } from '@patternfly/react-icons';
import { SortByDirection } from '@patternfly/react-table';
import findIndex from 'lodash/findIndex';
import propTypes from 'prop-types';
import React from 'react';
import { impactColorList, impactList, PUBLIC_DATE_OPTIONS } from './constants';
import qs from 'query-string';
import { IntlProvider } from '@redhat-cloud-services/frontend-components-translations';

export const dataShape = propTypes.shape({
    data: propTypes.oneOfType([propTypes.object, propTypes.array]),
    meta: propTypes.object,
    isLoading: propTypes.bool
});

/**
 * Based on the  allowedParams it will construct the arguments for the API call
 * which is defined in the vulnerabilities-client. vulnerabilities-client is written is typescript
 * so we need to define and pass every single parameter. Order in allowedParams is important
 *
 * @param {Object} apiProps
 * @param {Array} allowedParams - order should align with the params in the vulnerabilities-client
 */
export function constructParameters(apiProps, allowedParams) {
    if (apiProps) {
        Object.keys(apiProps).forEach(
            key => (
                apiProps[key] === undefined
                || apiProps[key] === ''
            )
            && delete apiProps[key]
        );

        const params = allowedParams.map(item => apiProps[item]);

        return params;
    }

    return [];
}

// TODO DRY:similar to constructParameters
export function constructURLParameters(urlParams, allowedParams) {
    if (urlParams) {
        const params = { ...urlParams };
        Object.keys(urlParams).forEach(
            key => (
                params[key] === undefined
                || params[key] === ''
                || !allowedParams.includes(key)
                || params[key] === false
            )
            && delete params[key]

        );
        return params;
    }
}

export function formatDate(date = new Date(), includeTime = false) {
    const prepend = (number) => `${`${number}`.length === 1 ? '0' : ''}${number}`;
    const toFormat = new Date(date);
    if (toFormat instanceof Date && !isNaN(toFormat)) {
        const year = toFormat.getFullYear();
        const month = toFormat.getMonth() + 1; // month is zero indexed
        const day = toFormat.getDate();
        const hour = toFormat.getUTCHours();
        const minute = toFormat.getUTCMinutes();

        return includeTime
            ? `${year}-${prepend(month)}-${prepend(day)} ${prepend(hour)}:${prepend(minute)} UTC`
            : `${year}-${prepend(month)}-${prepend(day)}`;
    }

    return toFormat;
}

/**
 * Contructs the filters which have more than one parameter that needs to be passed
 * and resets all the parameteres when it's needed
 *
 * @param {Object} filterParams
 * @return {Object}
 */
export function constructFilterParameters(filterParams) {
    if (Object.prototype.hasOwnProperty.call(filterParams, 'publish_date')) {
        let publicEntry = PUBLIC_DATE_OPTIONS.find(item => item.value === filterParams.publish_date);

        if (typeof publicEntry !== 'undefined') {
            filterParams.public_from = publicEntry.from && formatDate(publicEntry.from);
            filterParams.public_to = publicEntry.to && formatDate(publicEntry.to);
        } else {
            filterParams.public_from = undefined;
            filterParams.public_to = undefined;
            filterParams.publish_date = undefined;
        }
    }

    return filterParams;
}

export function getImpactDetails(impact) {
    if (Object.keys(impactList).includes(impact)) {
        return {
            title: impactList[impact].title,
            color: impactList[impact].color,
            text: impactList[impact].text,
            titleContent: (
                <SecurityIcon size="lg" />
            )
        };
    } else {
        return {
            title: 'Unknown',
            color: impactColorList.unknown,
            text: '',
            titleContent: (
                <UnknownIcon size="lg" />
            )
        };
    }
}

export const createSortBy = (header, value) => {
    if (value) {
        let direction = value.indexOf('-') === 0 ? SortByDirection.desc : SortByDirection.asc;
        value = value.replace(/^(-|\+)/, '');
        const index = findIndex(header, item => item.key === value);
        let sort = {
            index,
            direction
        };
        return sort;
    }

    return {};
};

export const handleSortColumn = (key, direction, header, currentSort, apply) => {
    let columnName = header[key].key;
    const useDefault = currentSort && currentSort.substr(1) !== columnName;
    if (direction === SortByDirection.desc || useDefault) {
        columnName = '-' + columnName;
    }

    apply({ sort: columnName });
};

export const updateStateSet = (stateSet, names, value) => {
    [].concat(names).forEach(name => {
        value ? stateSet.add(name) : stateSet.delete(name);
    });
    return stateSet;
};

export const useUrlParams = (allowedParams) => {
    const url = new URL(window.location);
    const urlParams = qs.parse(url.search);

    const setUrlParams = (parameters) => {
        const searchParams = qs.stringify(constructURLParameters(parameters, allowedParams));
        window.history.replaceState(null, null, `${url.origin}${url.pathname}?${searchParams}`);
    };

    return [urlParams, setUrlParams];
};

export const updateRef = (meta, apply) => {
    const pages = parseInt(meta.pages);
    const page = parseInt(meta.page);
    const cvesCount = parseInt(meta.cvesCount);
    const goTo = (pages === page && page > 1 && cvesCount === 1) ? (pages - 1) : page;

    apply({ page: goTo });
};

export const mountWithIntl = (Component) => {
    const wrapper = mount(Component, {
        wrappingComponent: IntlProvider
    });
    const provider = wrapper.getWrappingComponent();
    provider.setProps({ locale: 'en' });

    return wrapper;
};

export const shallowWithIntl = (Component) => {
    const wrapper = shallow(Component, {
        wrappingComponent: IntlProvider
    });
    const provider = wrapper.getWrappingComponent();
    provider.setProps({ locale: 'en' });

    return wrapper;
};

export const insertIf = (condition, ...elements) => {
    return condition ? elements : [];
};
