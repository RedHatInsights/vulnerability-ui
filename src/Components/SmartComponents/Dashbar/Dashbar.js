import React from 'react';
import { Card, Grid, GridItem, Split, SplitItem, StackItem, Stack, Alert, CardBody, Text } from '@patternfly/react-core';
import { Main } from '@redhat-cloud-services/frontend-components/Main';
import { SecurityIcon } from '@patternfly/react-icons';
import { CVES_ALLOWED_PARAMS } from '../../../Helpers/constants';
import { constructFilterParameters, useUrlParams } from '../../../Helpers/MiscHelper';
import { useDispatch } from 'react-redux';
import { changeCveListParameters } from '../../../Store/Actions/Actions';
import { FormattedMessage } from 'react-intl';
import messages from '../../../Messages';

const Dashbar = () => {
    const dispatch = useDispatch();
    const [urlParameters, setUrlParams] = useUrlParams([CVES_ALLOWED_PARAMS]);

    const apply = (filterParams = {}) => {
        setUrlParams(filterParams);
        const params = constructFilterParameters(filterParams);
        dispatch(changeCveListParameters(params));
    };

    return (urlParameters?.dashbar === 'true' &&
        <Main style={{ paddingBottom: 0 }}>
            <Stack hasGutter>
                <StackItem>
                    <Grid hasGutter>
                        <GridItem span={12} md={3}>
                            <Card isFullHeight className="card-box">
                                <CardBody>
                                    <Text>
                                        <span className="pf-u-font-size-2xl pf-u-font-weight-bold">
                                        5
                                        </span>
                                        <span className="pf-u-font-size-md pf-u-font-weight-light pf-u-ml-sm">
                                            <FormattedMessage {...messages.dashbarKnownExploitsTitle} />
                                        </span>
                                    </Text>
                                    <Text>
                                        <a onClick={() => apply({ ...urlParameters, known_exploit: 'true' })}>
                                            <FormattedMessage {...messages.dashbarKnownExploitsFilterText} />
                                        </a>
                                    </Text>
                                </CardBody>
                            </Card>
                        </GridItem>
                        <GridItem span={12} md={3}>
                            <Card isFullHeight>
                                <CardBody>
                                    <Text>
                                        <span className="pf-u-font-size-2xl pf-u-font-weight-bold">
                                        2
                                        </span>
                                        <span className="pf-u-font-size-md pf-u-font-weight-light pf-u-ml-sm">
                                            <FormattedMessage {...messages.dashbarSecurityRulesTitle} />
                                        </span>
                                    </Text>
                                    <Text>
                                        <a onClick={() => apply({ ...urlParameters, rule_presence: 'true' })}>
                                            <FormattedMessage {...messages.dashbarSecurityRulesFilterText} />
                                        </a>
                                    </Text>
                                </CardBody>
                            </Card>
                        </GridItem>
                        <GridItem span={12} md={6}>
                            <Card isFullHeight>
                                <CardBody>
                                    <Split hasGutter>
                                        <SplitItem isFilled>
                                            <Text>
                                                <SecurityIcon
                                                    color="var(--pf-global--danger-color--200)"
                                                    size="sm"
                                                    noVerticalAlign
                                                />
                                                <span className="pf-u-ml-sm pf-u-font-size-2xl pf-u-font-weight-bold">
                                                2
                                                </span>
                                                <span className="pf-u-font-size-md pf-u-font-weight-light pf-u-ml-sm">
                                                    <FormattedMessage {...messages.dashbarCriticalVulnerabilitiesTitle} />
                                                </span>
                                            </Text>
                                            <Text>
                                                <a onClick={() => apply({ ...urlParameters, impact: '7' })}>
                                                    <FormattedMessage {...messages.dashbarCriticalVulnerabilitiesFilterText} />
                                                </a>
                                            </Text>
                                        </SplitItem>
                                        <SplitItem isFilled>
                                            <Text>
                                                <SecurityIcon
                                                    color="var(--pf-global--warning-color--100)"
                                                    size="sm"
                                                    noVerticalAlign
                                                />
                                                <span className="pf-u-ml-sm pf-u-font-size-2xl pf-u-font-weight-bold">
                                                3
                                                </span>
                                                <span className="pf-u-font-size-md pf-u-font-weight-light pf-u-ml-sm">
                                                    <FormattedMessage {...messages.dashbarImportantVulnerabilitiesTitle} />
                                                </span>
                                            </Text>
                                            <Text>
                                                <a onClick={() => apply({ ...urlParameters, impact: '5' })}>
                                                    <FormattedMessage {...messages.dashbarImportantVulnerabilitiesFilterText} />
                                                </a>
                                            </Text>
                                        </SplitItem>
                                    </Split>
                                </CardBody>
                            </Card>
                        </GridItem>
                    </Grid>
                </StackItem>
                <StackItem>
                    <Alert
                        variant="warning"
                        isInline
                        title="This is an announcement"
                    >
                   Lorem ipsum
                    </Alert>
                </StackItem>
            </Stack>
        </Main>
    );
};

export default Dashbar;
