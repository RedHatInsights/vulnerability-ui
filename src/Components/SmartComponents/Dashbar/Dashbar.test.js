import Dashbar from './Dashbar';
import { Card, Grid } from '@patternfly/react-core';
import toJson from 'enzyme-to-json';
import { mountWithIntl } from '../../../Helpers/MiscHelper';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { BrowserRouter as Router } from 'react-router-dom';
window.insights = {};

jest.mock("react-redux", () => ({
    ...jest.requireActual("react-redux"),
    useSelector: jest.fn(),
    useDispatch: jest.fn()
}));

const mockStore = configureStore([store => next => action => { }]);
let store = mockStore({});

jest.mock('../../../Helpers/MiscHelper', () => ({
    ...jest.requireActual('../../../Helpers/MiscHelper'),
    useUrlParams: () => [{ dashbar: "true" }, jest.fn()]
}))

describe('Dashbar', () => {
    it('Should match the snapshot', () => {
        const wrapper = mountWithIntl(
            <Provider store={store}>
                <Router>
                    <Dashbar />
                </Router>
            </Provider>);
        expect(toJson(wrapper)).toMatchSnapshot();
    })
    it('Should render Grid with props hasGutter = true', () => {
        const wrapper = mountWithIntl(
            <Provider store={store}>
                <Router>
                    <Dashbar />
                </Router>
            </Provider>
        );
        expect(wrapper.find(Grid)).toHaveLength(1);
        expect(wrapper.find(Grid).prop('hasGutter')).toBeTruthy();
    });
    it('Should have 3 card items', () => {
        const wrapper = mountWithIntl(
            <Provider store={store}>
                <Router>
                    <Dashbar />
                </Router>
            </Provider>
        );
        expect(wrapper.find(Card)).toHaveLength(3);
    })
})