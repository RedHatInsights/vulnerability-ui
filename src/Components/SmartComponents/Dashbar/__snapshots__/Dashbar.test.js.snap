// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashbar Should match the snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Dashbar>
        <Connect(Main)
          style={
            Object {
              "paddingBottom": 0,
            }
          }
        >
          <Main
            style={
              Object {
                "paddingBottom": 0,
              }
            }
          >
            <section
              className="pf-l-page__main-section pf-c-page__main-section"
              page-type=""
              style={
                Object {
                  "paddingBottom": 0,
                }
              }
            >
              <Stack
                hasGutter={true}
              >
                <div
                  className="pf-l-stack pf-m-gutter"
                >
                  <StackItem>
                    <div
                      className="pf-l-stack__item"
                    >
                      <Grid
                        hasGutter={true}
                      >
                        <div
                          className="pf-l-grid pf-m-gutter"
                        >
                          <GridItem
                            md={3}
                            span={12}
                          >
                            <div
                              className="pf-l-grid__item pf-m-12-col pf-m-3-col-on-md"
                            >
                              <Card
                                className="card-box"
                                isFullHeight={true}
                              >
                                <article
                                  className="pf-c-card pf-m-full-height card-box"
                                  data-ouia-component-id="OUIA-Generated-Card-1"
                                  data-ouia-component-type="PF4/Card"
                                  data-ouia-safe={true}
                                  id=""
                                >
                                  <CardBody>
                                    <div
                                      className="pf-c-card__body"
                                    >
                                      <Text>
                                        <p
                                          className=""
                                          data-ouia-component-id="OUIA-Generated-Text-1"
                                          data-ouia-component-type="PF4/Text"
                                          data-ouia-safe={true}
                                          data-pf-content={true}
                                        >
                                          <span
                                            className="pf-u-font-size-2xl pf-u-font-weight-bold"
                                          >
                                            5
                                          </span>
                                          <span
                                            className="pf-u-font-size-md pf-u-font-weight-light pf-u-ml-sm"
                                          >
                                            <FormattedMessage
                                              defaultMessage="Known exploits"
                                              description="Title of Known exploit card"
                                              id="dashbarKnownExploitsTitle"
                                            >
                                              Known exploits
                                            </FormattedMessage>
                                          </span>
                                        </p>
                                      </Text>
                                      <Text>
                                        <p
                                          className=""
                                          data-ouia-component-id="OUIA-Generated-Text-2"
                                          data-ouia-component-type="PF4/Text"
                                          data-ouia-safe={true}
                                          data-pf-content={true}
                                        >
                                          <a
                                            onClick={[Function]}
                                          >
                                            <FormattedMessage
                                              defaultMessage="Filter by CVEs with known exploits"
                                              description="Link to Known exploits filter"
                                              id="dashbarKnownExploitsFilterText"
                                            >
                                              Filter by CVEs with known exploits
                                            </FormattedMessage>
                                          </a>
                                        </p>
                                      </Text>
                                    </div>
                                  </CardBody>
                                </article>
                              </Card>
                            </div>
                          </GridItem>
                          <GridItem
                            md={3}
                            span={12}
                          >
                            <div
                              className="pf-l-grid__item pf-m-12-col pf-m-3-col-on-md"
                            >
                              <Card
                                isFullHeight={true}
                              >
                                <article
                                  className="pf-c-card pf-m-full-height"
                                  data-ouia-component-id="OUIA-Generated-Card-2"
                                  data-ouia-component-type="PF4/Card"
                                  data-ouia-safe={true}
                                  id=""
                                >
                                  <CardBody>
                                    <div
                                      className="pf-c-card__body"
                                    >
                                      <Text>
                                        <p
                                          className=""
                                          data-ouia-component-id="OUIA-Generated-Text-3"
                                          data-ouia-component-type="PF4/Text"
                                          data-ouia-safe={true}
                                          data-pf-content={true}
                                        >
                                          <span
                                            className="pf-u-font-size-2xl pf-u-font-weight-bold"
                                          >
                                            2
                                          </span>
                                          <span
                                            className="pf-u-font-size-md pf-u-font-weight-light pf-u-ml-sm"
                                          >
                                            <FormattedMessage
                                              defaultMessage="Security Rules"
                                              description="Title of Security rule card"
                                              id="dashbarSecurityRulesTitle"
                                            >
                                              Security Rules
                                            </FormattedMessage>
                                          </span>
                                        </p>
                                      </Text>
                                      <Text>
                                        <p
                                          className=""
                                          data-ouia-component-id="OUIA-Generated-Text-4"
                                          data-ouia-component-type="PF4/Text"
                                          data-ouia-safe={true}
                                          data-pf-content={true}
                                        >
                                          <a
                                            onClick={[Function]}
                                          >
                                            <FormattedMessage
                                              defaultMessage="Filter by CVEs with security rules"
                                              description="Link to Security rules filter"
                                              id="dashbarSecurityRulesFilterText"
                                            >
                                              Filter by CVEs with security rules
                                            </FormattedMessage>
                                          </a>
                                        </p>
                                      </Text>
                                    </div>
                                  </CardBody>
                                </article>
                              </Card>
                            </div>
                          </GridItem>
                          <GridItem
                            md={6}
                            span={12}
                          >
                            <div
                              className="pf-l-grid__item pf-m-12-col pf-m-6-col-on-md"
                            >
                              <Card
                                isFullHeight={true}
                              >
                                <article
                                  className="pf-c-card pf-m-full-height"
                                  data-ouia-component-id="OUIA-Generated-Card-3"
                                  data-ouia-component-type="PF4/Card"
                                  data-ouia-safe={true}
                                  id=""
                                >
                                  <CardBody>
                                    <div
                                      className="pf-c-card__body"
                                    >
                                      <Split
                                        hasGutter={true}
                                      >
                                        <div
                                          className="pf-l-split pf-m-gutter"
                                        >
                                          <SplitItem
                                            isFilled={true}
                                          >
                                            <div
                                              className="pf-l-split__item pf-m-fill"
                                            >
                                              <Text>
                                                <p
                                                  className=""
                                                  data-ouia-component-id="OUIA-Generated-Text-5"
                                                  data-ouia-component-type="PF4/Text"
                                                  data-ouia-safe={true}
                                                  data-pf-content={true}
                                                >
                                                  <SecurityIcon
                                                    color="var(--pf-global--danger-color--200)"
                                                    noVerticalAlign={true}
                                                    size="sm"
                                                  >
                                                    <svg
                                                      aria-hidden={true}
                                                      aria-labelledby={null}
                                                      fill="var(--pf-global--danger-color--200)"
                                                      height="1em"
                                                      role="img"
                                                      style={null}
                                                      viewBox="0 0 896 1024"
                                                      width="1em"
                                                    >
                                                      <path
                                                        d="M861.5,0 L34.5,0 C15.4,0 0,14.3 0,32 L0,452.1 C0,768 387.7,1024 448.5,1024 C509.3,1024 896,768 896,452.2 L896,32 C896,14.3 880.6,0 861.5,0 Z M490.7,768 L405.3,768 C393.5,767.8 384.2,757.5 384,744.7 L384,663.3 C384.2,650.5 393.6,640.3 405.3,640 L490.7,640 C502.5,640.2 511.8,650.5 512,663.3 L512,744.7 L512.1,744.7 C511.8,757.5 502.4,767.8 490.7,768 Z M543.9,162.7 L517.2,514.4 C515.8,530.9 502,544 485.3,544 L410.6,544 C394,544 380.1,531.2 378.7,514.7 L352.1,163 C350.5,144.3 365.3,128.3 384,128.3 L512,128 C530.7,128 545.4,144 543.9,162.7 Z"
                                                      />
                                                    </svg>
                                                  </SecurityIcon>
                                                  <span
                                                    className="pf-u-ml-sm pf-u-font-size-2xl pf-u-font-weight-bold"
                                                  >
                                                    2
                                                  </span>
                                                  <span
                                                    className="pf-u-font-size-md pf-u-font-weight-light pf-u-ml-sm"
                                                  >
                                                    <FormattedMessage
                                                      defaultMessage="Critical vulnerabilities"
                                                      description="Title of Critical vulnerabilities card"
                                                      id="dashbarCriticalVulnerabilitiesTitle"
                                                    >
                                                      Critical vulnerabilities
                                                    </FormattedMessage>
                                                  </span>
                                                </p>
                                              </Text>
                                              <Text>
                                                <p
                                                  className=""
                                                  data-ouia-component-id="OUIA-Generated-Text-6"
                                                  data-ouia-component-type="PF4/Text"
                                                  data-ouia-safe={true}
                                                  data-pf-content={true}
                                                >
                                                  <a
                                                    onClick={[Function]}
                                                  >
                                                    <FormattedMessage
                                                      defaultMessage="Filter by CVEs with critical severity"
                                                      description="Link to Critical vulnerabilities filter"
                                                      id="dashbarCriticalVulnerabilitiesFilterText"
                                                    >
                                                      Filter by CVEs with critical severity
                                                    </FormattedMessage>
                                                  </a>
                                                </p>
                                              </Text>
                                            </div>
                                          </SplitItem>
                                          <SplitItem
                                            isFilled={true}
                                          >
                                            <div
                                              className="pf-l-split__item pf-m-fill"
                                            >
                                              <Text>
                                                <p
                                                  className=""
                                                  data-ouia-component-id="OUIA-Generated-Text-7"
                                                  data-ouia-component-type="PF4/Text"
                                                  data-ouia-safe={true}
                                                  data-pf-content={true}
                                                >
                                                  <SecurityIcon
                                                    color="var(--pf-global--warning-color--100)"
                                                    noVerticalAlign={true}
                                                    size="sm"
                                                  >
                                                    <svg
                                                      aria-hidden={true}
                                                      aria-labelledby={null}
                                                      fill="var(--pf-global--warning-color--100)"
                                                      height="1em"
                                                      role="img"
                                                      style={null}
                                                      viewBox="0 0 896 1024"
                                                      width="1em"
                                                    >
                                                      <path
                                                        d="M861.5,0 L34.5,0 C15.4,0 0,14.3 0,32 L0,452.1 C0,768 387.7,1024 448.5,1024 C509.3,1024 896,768 896,452.2 L896,32 C896,14.3 880.6,0 861.5,0 Z M490.7,768 L405.3,768 C393.5,767.8 384.2,757.5 384,744.7 L384,663.3 C384.2,650.5 393.6,640.3 405.3,640 L490.7,640 C502.5,640.2 511.8,650.5 512,663.3 L512,744.7 L512.1,744.7 C511.8,757.5 502.4,767.8 490.7,768 Z M543.9,162.7 L517.2,514.4 C515.8,530.9 502,544 485.3,544 L410.6,544 C394,544 380.1,531.2 378.7,514.7 L352.1,163 C350.5,144.3 365.3,128.3 384,128.3 L512,128 C530.7,128 545.4,144 543.9,162.7 Z"
                                                      />
                                                    </svg>
                                                  </SecurityIcon>
                                                  <span
                                                    className="pf-u-ml-sm pf-u-font-size-2xl pf-u-font-weight-bold"
                                                  >
                                                    3
                                                  </span>
                                                  <span
                                                    className="pf-u-font-size-md pf-u-font-weight-light pf-u-ml-sm"
                                                  >
                                                    <FormattedMessage
                                                      defaultMessage="Important vulnerabilities"
                                                      description="Title of Important vulnerabilities card"
                                                      id="dashbarImportantVulnerabilitiesTitle"
                                                    >
                                                      Important vulnerabilities
                                                    </FormattedMessage>
                                                  </span>
                                                </p>
                                              </Text>
                                              <Text>
                                                <p
                                                  className=""
                                                  data-ouia-component-id="OUIA-Generated-Text-8"
                                                  data-ouia-component-type="PF4/Text"
                                                  data-ouia-safe={true}
                                                  data-pf-content={true}
                                                >
                                                  <a
                                                    onClick={[Function]}
                                                  >
                                                    <FormattedMessage
                                                      defaultMessage="Filter by CVEs with important severity"
                                                      description="Link to Important vulnerabilities filter"
                                                      id="dashbarImportantVulnerabilitiesFilterText"
                                                    >
                                                      Filter by CVEs with important severity
                                                    </FormattedMessage>
                                                  </a>
                                                </p>
                                              </Text>
                                            </div>
                                          </SplitItem>
                                        </div>
                                      </Split>
                                    </div>
                                  </CardBody>
                                </article>
                              </Card>
                            </div>
                          </GridItem>
                        </div>
                      </Grid>
                    </div>
                  </StackItem>
                  <StackItem>
                    <div
                      className="pf-l-stack__item"
                    >
                      <Alert
                        isInline={true}
                        title="This is an announcement"
                        variant="warning"
                      >
                        <div
                          aria-label="Warning Alert"
                          className="pf-c-alert pf-m-inline pf-m-warning"
                          data-ouia-component-id="OUIA-Generated-Alert-warning-1"
                          data-ouia-component-type="PF4/Alert"
                          data-ouia-safe={true}
                          onMouseEnter={[Function]}
                          onMouseLeave={[Function]}
                        >
                          <AlertIcon
                            variant="warning"
                          >
                            <div
                              className="pf-c-alert__icon"
                            >
                              <ExclamationTriangleIcon
                                color="currentColor"
                                noVerticalAlign={false}
                                size="sm"
                              >
                                <svg
                                  aria-hidden={true}
                                  aria-labelledby={null}
                                  fill="currentColor"
                                  height="1em"
                                  role="img"
                                  style={
                                    Object {
                                      "verticalAlign": "-0.125em",
                                    }
                                  }
                                  viewBox="0 0 576 512"
                                  width="1em"
                                >
                                  <path
                                    d="M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
                                  />
                                </svg>
                              </ExclamationTriangleIcon>
                            </div>
                          </AlertIcon>
                          <h4
                            className="pf-c-alert__title"
                          >
                            <span
                              className="pf-u-screen-reader"
                            >
                              Warning alert:
                            </span>
                            This is an announcement
                          </h4>
                          <div
                            className="pf-c-alert__description"
                          >
                            Lorem ipsum
                          </div>
                        </div>
                      </Alert>
                    </div>
                  </StackItem>
                </div>
              </Stack>
            </section>
          </Main>
        </Connect(Main)>
      </Dashbar>
    </Router>
  </BrowserRouter>
</Provider>
`;
